<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
               "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml"
lang="en" xml:lang="en">
<head>
<title>IMA: An R package for high-throughput analysis of Illumina’s 450K Infinium methylation data.</title>
<meta http-equiv="Content-Type" content="text/html;charset=utf-8"/>
<meta name="generator" content="Org-mode"/>
<meta name="generated" content="2011-12-07 15:03:26 EST"/>
<meta name="author" content="Dan Wang   "/>
<meta name="description" content=""/>
<meta name="keywords" content=""/>
<style type="text/css">
 <!--/*--><![CDATA[/*><!--*/
  html { font-family: Times, serif; font-size: 12pt; }
  .title  { text-align: center; }
  .todo   { color: red; }
  .done   { color: green; }
  .tag    { background-color: #add8e6; font-weight:normal }
  .target { }
  .timestamp { color: #bebebe; }
  .timestamp-kwd { color: #5f9ea0; }
  p.verse { margin-left: 3% }
  pre {
	border: 1pt solid #AEBDCC;
	background-color: #F3F5F7;
	padding: 5pt;
	font-family: courier, monospace;
        font-size: 90%;
        overflow:auto;
  }
  table { border-collapse: collapse; }
  td, th { vertical-align: top; }
  dt { font-weight: bold; }
  div.figure { padding: 0.5em; }
  div.figure p { text-align: center; }
  .linenr { font-size:smaller }
  .code-highlighted {background-color:#ffff00;}
  .org-info-js_info-navigation { border-style:none; }
  #org-info-js_console-label { font-size:10px; font-weight:bold;
                               white-space:nowrap; }
  .org-info-js_search-highlight {background-color:#ffff00; color:#000000;
                                 font-weight:bold; }
  /*]]>*/-->
</style>
<script type="text/javascript">
<!--/*--><![CDATA[/*><!--*/
 function CodeHighlightOn(elem, id)
 {
   var target = document.getElementById(id);
   if(null != target) {
     elem.cacheClassElem = elem.className;
     elem.cacheClassTarget = target.className;
     target.className = "code-highlighted";
     elem.className   = "code-highlighted";
   }
 }
 function CodeHighlightOff(elem, id)
 {
   var target = document.getElementById(id);
   if(elem.cacheClassElem)
     elem.className = elem.cacheClassElem;
   if(elem.cacheClassTarget)
     target.className = elem.cacheClassTarget;
 }
/*]]>*///-->
</script>
</head>
<body>
<div id="content">

<h1 class="title">IMA: An R package for high-throughput analysis of Illumina’s 450K Infinium methylation data.</h1>


<p><font color = "red">***The package is maintained on <a href="#rforge.net">rforge.net</a>: <a href="https://www.rforge.net/IMA/">https://www.rforge.net/IMA/</a></font><br>
</p>
<div id="table-of-contents">
<h2>Table of Contents</h2>
<div id="text-table-of-contents">
<ul>
<li><a href="#sec-1">1 Abstract </a></li>
<li><a href="#sec-2">2 Installation </a></li>
<li><a href="#sec-3">3 Tutorial </a></li>
<li><a href="#sec-4">4 Annotation file </a></li>
<li><a href="#sec-5">5 Pipeline </a></li>
<li><a href="#sec-6">6 Citations </a></li>
<li><a href="#sec-7">7 Frequently Asked Questions: </a></li>

</ul>
</div>
</div>

<div id="outline-container-1" class="outline-2">
<h2 id="sec-1"><span class="section-number-2">1</span> Abstract </h2>
<div class="outline-text-2" id="text-1">

<p>The Illumina Infinium HumanMethylation450 BeadChip is a newly designed high-density microarray for quantifying the methylation level of over 450,000 CpG sites within human genome. IMA (Illumina Methylation Analyzer) is a computational package designed to automate the pipeline for analyzing site-level and region-level methylation changes in epigenetic studies utilizing the 450K DNA methylation microarray. The pipeline loads the data from Illumina platform and provides user-customized functions commonly required to perform exploratory methylation analysis and summarization for individual sites as well as annotated regions.
</p>
<p>
Note that instead of providing recommendations about which specific analysis method should be used, the main purpose of developing IMA package is to provide a range of commonly used Infinium methylation microarray analysis options for users to choose for their exploratory analysis and summarization in an automatic way. Therefore, it is the best interest for the users to consult experienced bioinformatician/statistician about which specific analysis option/route should be chosen for their 450k microarray data.
</p>
</div>

</div>

<div id="outline-container-2" class="outline-2">
<h2 id="sec-2"><span class="section-number-2">2</span> Installation </h2>
<div class="outline-text-2" id="text-2">

<p><font color="red">***Prerequisites: </font><br>
</p>
<p>
1.The IMA package requires R version &gt;= 2.13 for Windows system, and R version &gt;=2.11.0 for Linux like system. 
</p>
<p>
2.The IMA package requires the following packages to be installed: WriteXLS,limma,MASS,bioDist,dplR.
If your system does not have them installed, the easiest way to install them is to issue the following command at the R prompt:
</p>
<p>
source("<a href="http://bioconductor.org/biocLite.R">http://bioconductor.org/biocLite.R</a>");
biocLite(c("limma","bioDist"));
install.packages(c("WriteXLS","MASS"),repos="<a href="http://cran.r-project.org">http://cran.r-project.org</a>");
</p>
<p>
3.The WriteXLS package requires the perl to be installed.
</p>

<p>
**Install options:There are two ways to install IMA package: 
</p>
<p>
Option 1): Issue the following command at the R prompt:
</p>
<p>
install.packages("IMA",repos=c("<a href="http://rforge.net">http://rforge.net</a>"))
</p>
<p>
Option 2): download the package <a href="./IMA_3.1.2.tar.gz">here</a> and issue the following command at the R prompt:
</p>
<p>
install.packages("IMA_3.1.2.tar.gz",repos=NULL,type = "source")
</p>

</div>

</div>

<div id="outline-container-3" class="outline-2">
<h2 id="sec-3"><span class="section-number-2">3</span> Tutorial </h2>
<div class="outline-text-2" id="text-3">

<p>A vignentte that illustrate various aspects of IMA is available <a href="./meth450k.pdf">here</a>
</p>
<p>
The user manual of IMA package could be found <a href="./IMA-manual.pdf">here</a>.
</p></div>

</div>

<div id="outline-container-4" class="outline-2">
<h2 id="sec-4"><span class="section-number-2">4</span> Annotation file </h2>
<div class="outline-text-2" id="text-4">

<p>
The region-level annotation library for the 450k microarray could be produced by issue below code after read in your raw methylation data into R. 
</p>
<p>
&gt;dataf2 = IMA.methy450PP(data,peakcorrection = FALSE,na.omit = FALSE,normalization=FALSE,transfm = FALSE,samplefilterdetectP =FALSE,locidiff = FALSE, XYchrom = FALSE,snpfilter=FALSE )
</p>
<p>
&gt;fullannot = dataf2@annot
</p>
<p>
&gt;temp = c("TSS1500Ind","TSS200Ind","UTR5Ind", "EXON1Ind","GENEBODYInd","UTR3Ind","ISLANDInd","NSHOREInd","SSHOREInd","NSHELFInd", "SSHELFInd")
</p>
<p>
&gt;for( i in 1:11){eval(parse(text=paste(temp[i],"=dataf2@",temp[i],sep="")))}
</p>
<p>
&gt;eval(parse(text = paste("save(fullannot", paste(temp,collapse = ","), "file = 'fullannotInd.rda')", sep = "," )))
</p>
Instead, the region-level annotation library for the 450k microarray could be downloaded from <a href="./fullannotInd.rda">here</a>.
</p>
<p>
Then users can load the regional-level annotation library by issuing the following command at the R prompt:
</p>
<p>
&gt;load("./fullannotInd.rda")
</p>
<p>
It is recommended to produce the annotation library by using user's own data, as it is very likely that different users may have slightly different annotation produced by GenomeStudio .

</div>

</div>

<div id="outline-container-5" class="outline-2">
<h2 id="sec-5"><span class="section-number-2">5</span> Pipeline </h2>
<div class="outline-text-2" id="text-5">


<p>
The <a href="./pipeline.R">pipeline</a> loads the data from Illumina platform and provides user-customized functions commonly required to perform differential methylation analysis nd summarization for individual sites as well as annotated regions. The user can either run the pipeline with default setting or specify optional routes in the parameter file. Note that it is the best interest for the users to consult experienced bioinformatician/statistician about which specific analysis option should be chosen for their 450k microarray data. 
</p>
<p>
To run the pipeline file, users can simply type "R no save &lt; pipeline.R" at the Linux/Unix prompt. Alternatively, users can copy the commands in the pipeline.R and paste them into the R prompt. 
</p></div>

</div>

<div id="outline-container-6" class="outline-2">
<h2 id="sec-6"><span class="section-number-2">6</span> Citations </h2>
<div class="outline-text-2" id="text-6">

<p>Wang D, Yan L, Hu Q, Sucheston LE, Higgins MJ, Ambrosone CB, Johnson CS, Smiraglia DJ, Liu S. IMA: an R package for high-throughput analysis of Illumina's 450K Infinium methylation data. Bioinformatics. 2012 Mar 1;28(5):729-30
</p></div>

</div>

<div id="outline-container-7" class="outline-2">
<h2 id="sec-7"><span class="section-number-2">7</span> Frequently Asked Questions </h2>
<div class="outline-text-2" id="text-7">

<p>1.There are a total of ~65k probes on the 450k platform which contain SNPs at/near the target CpG site and are unlikely to measure DNA methylation at all. Should this issue be considered?
</p>
<p>
Answer:
</p>
<p>
Users can choose to filter out loci whose methylation level are measured by probes containing SNP(s) at/near the targeted CpG site. We have included an optional route for users to filter out these SNP-containing probes in Version 2.1.0 or above. The list of SNP-containing probes (based on dbSNP v132) was provided by Ali Torkamani at Scripps Institute and could be downloaded from <a href="./snpsites.txt">here</a> or by issuing the following command in R :
</p>
<p>
&gt;snpfilter = system.file("extdata/snpsites.txt",package ="IMA").
</p>
<p>
2.I need to make a paired analysis for the samples and usually I would adjust for this using block or some other factor in LIMMA. However, I do not really see where I can add that type of info now. So if possible, I would really appreciate some info on this, otherwise, can the object be run outside the package as an input to LIMMA?
</p>
<p>
Answer:
</p>
<p>
We have included optional routes for paired analysis in Version 2.1.0 or above.
</p>
<p>
3.I found a list of Island regions differentially expressed using IMA testfunc. However, the results only contain the chromosome regions instead of ProbeID within the differentially expressed regions. Would it be possible to have some options to get the Probe Ids and their corresponding annotations within these regions?
</p>
<p>
Answer:
</p>
<p>
An example for how to extract probeID and corresponding annotation information within the differentially expressed region(s) has been added to the Vignette.
<p>4. Have you consider the peark correction in the data preprocessing step?
</p>
 <p>
Answer:
</p>
<p>
The peak correction option had been added to the preprocessing step. For the detail of the peak correction method, please reference "Evaluation of the Infinium Methylation 450k technology" by Sarah Dedeurwaerder et al. We fixed the bug after version 3.1.2 in the peak correction option, please use the latest version of IMA(>3.1.2) if you choose the peak correction option = TRUE.
<p>5. I have some data generated by the Methyl27k arrays. Can I use IMA for as well?
</p>
 <p>
Answer:
</p>
<p>
To make IMA configurable for 27k array, we first mapped the loci's annotation for 27k array to that of 450k array. There are a total of 27578 loci for 27k array, and 1600 of them couldn't be mapped to 450k array. For those unmapped loci, we keep their original annotation from the 27k array. For those mapped loci, we use the annotation from 450k array.
The annotation for 27k array could be downloaded from <a href="./annot27k_mapped.Rdata">here</a>. The usage of IMA to 27k array is similar to that for 450k array, except that the following two commands needs to be issued between the reading step(IMA.methy450R) and the preprocessing step(IMA.methy450PP):
</p>
 <p>
load("./annot27k_mapped.Rdata")
</p>
 <p>
data@annot = as.matrix(annotout)
</p></div>
</div>
<div id="postamble">
<p class="author"> Author: Dan Wang   
<a href="mailto:dan.wang@roswellpark.org ">&lt;dan.wang@roswellpark.org &gt;</a>
</p>
<p class="date"> Date: 2012-5-20 15:03:26 EST</p>
<p class="creator">HTML generated by org-mode 6.34c in emacs 23</p>
</div>
</div>
</body>
</html>
